staging/src/k8s.io/apiserver/pkg/server/filters/timeout.go

```diff
@@ -138,8 +138,8 @@ func (t *timeoutHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			}()
 		}()
 
-		postTimeoutFn()
 		tw.timeout(err)
```

Comment:
the `timeout` call can panic; moving `postTimeoutFn` after it means it will not get called in that case

IF we want to invoke `postTimeoutFn` after `tw.timeout` (I haven't paged this back in to know if that's correct or not), we should probably do that with a defer instead

Reply:
Yes, I overlooked it in haste.

But it seems right for the `postTimeoutFn` function to be invoked only after we mark the request as `it has timed out`. 
Today `postTimeoutFn` records a metric only.
```
		postTimeoutFn := func() {
			metrics.RecordRequestTermination(req, requestInfo, metrics.APIServerComponent, http.StatusGatewayTimeout)
		}
```